controllers/capabilities/tenant_status_reconciler.go

```diff
@@ -0,0 +1,100 @@
+package controllers
+
+import (
+	"context"
+
+	capabilitiesv1alpha1 "github.com/3scale/3scale-operator/apis/capabilities/v1alpha1"
+	"github.com/3scale/3scale-operator/pkg/reconcilers"
+
+	"github.com/3scale/3scale-operator/pkg/apispkg/common"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type TenantStatusReconciler struct {
+	*reconcilers.BaseReconciler
+	resourceUpdated  *capabilitiesv1alpha1.Tenant
+	resourceOriginal *capabilitiesv1alpha1.Tenant
+	reconcileError   error
+	logger           logr.Logger
+}
+
+func NewTenantStatusReconciler(b *reconcilers.BaseReconciler, resourceUpdated, resourceOriginal *capabilitiesv1alpha1.Tenant, reconcileError error) *TenantStatusReconciler {
+	return &TenantStatusReconciler{
+		BaseReconciler:   b,
+		resourceUpdated:  resourceUpdated,
+		resourceOriginal: resourceOriginal,
+		reconcileError:   reconcileError,
+		logger:           b.Logger().WithValues("Status Reconciler", resourceUpdated.Name),
+	}
+}
+
+// Compare and update the status of the tenant CR (both, object and object.status)
+func (s *TenantStatusReconciler) Reconcile() (reconcile.Result, error) {
+	// Check for changes to spec or metadata
+	equalSpec := s.resourceOriginal.SpecEqual(s.resourceUpdated, s.logger)
+	if !equalSpec {
+		s.logger.Info("updating tenant object")
+		err := s.Client().Update(context.TODO(), s.resourceUpdated)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	// Check for changes to the status
+	newStatus := s.calculateStatus()
+	equalStatus := s.resourceOriginal.Status.StatusEqual(&newStatus, s.logger)
+
+	if !equalStatus {
+		s.logger.Info("updating tenant status")
+		s.resourceUpdated.Status = newStatus
+		err := s.UpdateResourceStatus(s.resourceUpdated)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (s *TenantStatusReconciler) calculateStatus() capabilitiesv1alpha1.TenantStatus {
```

Comment:
alternatively - we could only have single condition - ready or not (not = error/in progress)

Reply:
My take is that we have a single condition. The tenant can be ready right after being created. I would have "Ready" or "Available" only. Their state can be `false` and still you have `message` and `reason` to specify further what is going on. For instance if the user input is required (like when the secret lacks some field)

Reply:
Updated, thank you